news/2020-02-03-Micro_Profile_OpenTracing_Comes_To_WildFly.adoc

```diff
@@ -0,0 +1,113 @@
+[source,ruby]
+--
+[standalone@localhost:9990 /] /socket-binding-group=standard-sockets/remote-destination-outbound-socket-binding=jaeger:add(host=localhost, port=6831)
+{"outcome" => "success"}
+--
+
+Now we can define our MPOT tracer configuration:
+[source,ruby]
+--
+[standalone@localhost:9990 /] /subsystem=microprofile-opentracing-smallrye/jaeger-tracer=jaeger-demo:add(sampler-type=const, sampler-param=1, reporter-log-spans=true, sender-binding=jaeger)
+{"outcome" => "success"}
+--
+
+=== Using TCP
+
+If you want to use TCP instead of UDP you need to configure the *sender-endpoint* and set its value to *http://localhost:14268/api/traces*.
+
+[source,ruby]
+--
+[standalone@localhost:9990 /] /subsystem=microprofile-opentracing-smallrye/jaeger-tracer=jaeger:write-attribute(name="sender-endpoint", value="http://localhost:14268/api/traces")
+{
+    "outcome" => "success",
+    "response-headers" => {
+        "operation-requires-reload" => true,
+        "process-state" => "reload-required"
+    }
+}
+[standalone@localhost:9990 /] reload
+--
+
+=== Setting the default tracer
+
+Let's define this new tracer as the default tracer to be used by WildFly:
+[source,ruby]
+--
+[standalone@localhost:9990 /] /subsystem=microprofile-opentracing-smallrye:write-attribute(name=default-tracer, value=jaeger-demo)
+{
+    "outcome" => "success",
+    "response-headers" => {
+        "operation-requires-reload" => true,
+        "process-state" => "reload-required"
+    }
+}
+[standalone@localhost:9990 /] reload
+--
+
+[NOTE]
+====
+Instead of relying on the default tracer attribute, you can use the *web.xml* deployment descriptor and set the init-parameter *smallrye.opentracing.tracer.configuration* to the name of tracer to use.
+====
+
+== Deploying a demo application
+
+We provide a sample application at link:https://github.com/ehsavoie/opentracing-demo[https://github.com/ehsavoie/opentracing-demo].
+This application is a simple JAXRS endpoint producing XML or JSON and failing randomly.
+
+[source,bash]
+--
+git clone git@github.com:ehsavoie/opentracing-demo.git
+cd opentracing-demo
+mvn clean install
+cp target/opentracing-demo.war $WILDFLY_HOME/standalone/deployments
+$WILDFLY_HOME/bin/standalone.sh
+--
+
+Now, you can navigate to link:http://localhost:8080/opentracing-demo[http://localhost:8080/opentracing-demo] to access the deployed application. Generate a few traces by clicking on the links several times.
+
+Now open the link:http://localhost:16686[Jaeger UI] and search for traces related to the service *opentracing-demo.war*, you should see the result of your previous actions.
+
+image::opentracing/jaegertraces.png[]
+
+== References
+
+* WildFly Microprofile OpenTracing configuration: link:https://docs.wildfly.org/19/Admin_Guide.html#MicroProfile_OpenTracing_SmallRye[Admin Guide]
+* WildScribe: http://wildscribe.github.io/WildFly/19.0/subsystem/microprofile-opentracing-smallrye/jaeger-tracer/index.html
```

Comment:
This is now https://docs.wildfly.org/19/wildscribe/subsystem/microprofile-opentracing-smallrye/jaeger-tracer/index.html